platforms/access-bridge/src/controllers/checkout-controller.ts

```diff
@@ -63,4 +63,62 @@ export class CheckoutController {
       next(error);
     }
   }
+
+  /**
+   * Service handler for generating a Stripe Billing portal session based on the customer.
+   * @param req - Express request object
+   * @param res - Express response object
+   * @param next - Express next middleware function
+   * @returns A Promise that resolves with a response containing the URL for the Stripe Billing Portal session.
+   */
+  async generateBillingPortalURL(req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const siteId = req.params.site_id;
+      if (!isValidSiteId(siteId)) {
+        sendErrors(res, ErrorDefinitions.ParameterInvalidError.create({ parameterName: 'site_id' }));
+        return;
+      }
+
+      const authorization = req.headers['authorization'];
+      if (!authorization) {
+        sendErrors(res, ErrorDefinitions.UnauthorizedError.create());
+        return;
+      }
+
+      // Get the email address from the Authorization token
+      const viewer = await this.identityService.getAccount({ authorization });
+      if (!viewer.id || !viewer.email) {
+        sendErrors(res, ErrorDefinitions.UnauthorizedError.create());
+        return;
+      }
```

Comment:
We are doing this exact same snippet in 3 places now. It seems to be pretty central to the main flows of the auth bridge. Would it make sense to put it in an Express middleware so that the downstream methods can just rely on the request context having a viewer property populated?

Reply:
Agreed. The site id is moved to middleware. Not sure about the authorization as it's only mandatory here in the checkout controller. If makes sense later we can move it to a middleware as well.